frontend/src/services/settings.ts

```diff
@@ -66,39 +76,68 @@ export const maybeMigrateSettings = (logout: () => void) => {
 export const getDefaultSettings = (): Settings => DEFAULT_SETTINGS;
 
 /**
- * Get the settings from local storage or use the default settings if not found
+ * Get the settings from the server or use the default settings if not found
  */
-export const getSettings = (): Settings => {
-  const model = localStorage.getItem("LLM_MODEL");
+export const getSettings = async (): Promise<Settings> => {
```

Comment:
If this is called within a `useQuery` hook,  it may be better to _not_ wrap the logic in a `try...catch` block since `useQuery` will handle the error internally and allow us better control. Alternatively, instead of logging the error, we can throw a more descriptive one

https://tanstack.com/query/v4/docs/framework/react/guides/query-functions#handling-and-throwing-errors

Reply:
Good call - I've removed the try / catch from here so we can handle errors using tanstack